Enterprise-CMCS/macpro-mdct-mfp
# [CMDCT-3087] Cleanup backend data of no longer applicable target populations

### Description

If a user was to speed ahead and define an initiative, then check a population as part of that initiative, and then back track and mark a population as not applicable... Well, the data would live on in the initiative. This would mean it could accidentally get passed to the SAR or a new Work Plan when its copied over. So we need to clean up after this possible scenario! And wouldn't ya know it, I've done just that in this here PR :) 


https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/19439679/a4f66186-6e61-4f3c-8133-2e615c558bdb



### Related ticket(s)


CMDCT-3087

---
### How to test

Create a Work Plan
Create an initiative
Go to the define initiative page
Checkmark the first couple populations
Go to the Transition Benchmarks Page
Mark the Older Populations as not applicable (Or any you want, really)
See that the data you get back (By checking the preview tab and then looking at FieldData -> Initiative) and see that only the target populations that you didn't just mark as not applicable are still there

---
### Author checklist
